internal/services/kusto/kusto_cosmosdb_data_connection_resource_test.go

```diff
@@ -0,0 +1,275 @@
+package kusto_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/kusto/2022-12-29/dataconnections"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type KustoCosmosDBDataConnectionResource struct{}
+
+func (k KustoCosmosDBDataConnectionResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := dataconnections.ParseDataConnectionID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Kusto.DataConnectionsClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %v", id.String(), err)
+	}
+
+	if resp.Model != nil {
+		value, ok := (*resp.Model).(dataconnections.CosmosDbDataConnection)
+		if !ok {
+			return nil, fmt.Errorf("%s is not an CosmosDB Data Connection", id.String())
+		}
+		exists := value.Properties != nil
+		return &exists, nil
+	} else {
+		return nil, fmt.Errorf("response model is empty")
+	}
+}
+
+func TestAccKustoCosmosDBDataConnection_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kusto_cosmosdb_data_connection", "test")
+	r := KustoCosmosDBDataConnectionResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("cosmosdb_account_id").Exists(),
+				check.That(data.ResourceName).Key("cosmosdb_database").Exists(),
+				check.That(data.ResourceName).Key("cosmosdb_container").Exists(),
+				check.That(data.ResourceName).Key("cluster_name").Exists(),
+				check.That(data.ResourceName).Key("database_name").Exists(),
+				check.That(data.ResourceName).Key("table_name").Exists(),
+				check.That(data.ResourceName).Key("managed_identity_id").Exists(),
+				check.That(data.ResourceName).Key("table_name").HasValue("TestTable"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccKustoCosmosDBDataConnection_complete(t *testing.T) {
```

Comment:
Can you please add an update test as well?

Reply:
I have tested the update and it results in confusing error. So I did not add an update function in this resource.